src/client/types.ts

```diff
@@ -118,3 +118,22 @@ export type SelectorEngine = {
 
 export type RemoteAddr = channels.RemoteAddr;
 export type SecurityDetails = channels.SecurityDetails;
+
+export type NewRequestOptions = {
+  baseURL?: string;
+  extraHTTPHeaders?: { [key: string]: string; };
+  httpCredentials?: {
+    username: string;
+    password: string;
+  };
+  ignoreHTTPSErrors?: boolean;
+  proxy?: {
+    server: string;
+    bypass?: string;
+    username?: string;
+    password?: string;
+  };
+  timeout?: number;
+  userAgent?: string;
+  storageState?: string | channels.BrowserNewContextOptions['storageState'];
```

Comment:
Should this be `channels.PlaywrightNewRequestOptions` instead? Also, there is `StorageState` defined above in this file.

Reply:
Changed to StorageState.